packages/app/src/actions/user.js

```diff
@@ -1,6 +1,30 @@
-import { RECEIVED_USER } from '../reducers/user'
+import { RECEIVED_USER, SET_USER_AUTHORITY } from '../reducers/user'
+import { onError } from './index'
+
+import { apiFetchUserAuthority } from '../api/user'
 
 export const acReceivedUser = value => ({
     type: RECEIVED_USER,
     value,
 })
+
+export const acSetUserAuthority = value => ({
+    type: SET_USER_AUTHORITY,
+    value,
+})
+
+export const tLoadUserAuthority = authorityKey => async (
+    dispatch,
+    getState,
+    engine
+) => {
+    const onSuccess = response => {
+        dispatch(acSetUserAuthority({ [authorityKey]: response.authority }))
+    }
+
+    try {
+        return onSuccess(await apiFetchUserAuthority(engine, authorityKey))
+    } catch (err) {
+        onError('tLoadUserAuthority', err)
```

Comment:
What sort of error does this throw and should it be covered by our [error module](https://github.com/dhis2/data-visualizer-app/blob/master/packages/app/src/modules/error.js)?

Reply:
The fetch is done via the data engine, i guess generic errors can happen.
Not sure there are specific errors for that request, that would be a question for backend.

Reply:
Ah ok, actually I wasn't referring to using a specific error (which would also be good of course), but if this path would trigger `acSetLoadError()`? If not you can catch it and use `acSetLoadError(new GenericServerError())` to display a user friendly error.

Reply:
ok, done.